### PR TITLE
[feat] support graceful shutdown in metasrv

### LINKED ISSUE #2543: `databend-meta` graceful shutdown when killed
**Summary**

When `kill -9 <databend-meta pid>`, it always performance as:
```
bohu      114085  0.1  0.0      0     0 pts/5    Z+   12:18   0:11 [databend-meta] <defunct>
```
and never exit.

### PR DESCRIPTION
**Pull request by sunisdown**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

use stoppable in metasrv

## Changelog

- Improvement

## Related Issues

Resolve #2543

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: metasrv/src/api/http_service_test.rs [hunk 1]
```diff
@@ -29,7 +30,7 @@ use crate::tests::tls_constants::TEST_SERVER_KEY;
 #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
 async fn test_http_service_tls_server() -> Result<()> {
     let mut conf = Config::empty();
-    let addr_str = "127.0.0.1:0";
+    let addr_str = "127.0.0.1:40001".to_string();
```
Comment (zhang2014) [inline metasrv/src/api/http_service_test.rs:33]: Why aren't we using random port?

**Comment by zhang2014** [comment]
Look good to me. CC: @drmingdrmer

**Comment by sunisdown** [comment]
The lifetime and async in Rust are too difficult, I need to read the document again and learn how to write  Rust code.
Please help to comment as more as possible when reviewing it. thanks :) I have no confidence for this PR

File: metasrv/src/api/http_service_test.rs (hunk 1 above)
Comment (sunisdown) [reply metasrv/src/api/http_service_test.rs:33]: > Why aren't we using random port?

Let me change to random port and test it again, the test hung when I was coding  for this issue so I changed to a fixed port and forget to change it back.

**Review by drmingdrmer — APPROVED** [review]
Smells really good! Thank you man!

File: metasrv/src/api/http_service_test.rs (hunk 1 above)
Comment (sunisdown) [reply metasrv/src/api/http_service_test.rs]: @zhang2014   updated the test, but I still don't understand why the test runs fine in tokio when I bind port 0 and fails without tokio:
failed with `127.0.0.1:0`
```
        srv.start().await.expect("HTTP: admin api error");
        // kick off
        let client = reqwest::Client::builder()
            .add_root_certificate(cert)
            .build()
            .unwrap();
        let resp = client.get(url).send().await;
        assert!(resp.is_ok());
        let resp = resp.unwrap();
        assert!(resp.status().is_success());
        assert_eq!("/v1/health", resp.url().path());
```
works with `127.0.0.1:`
```
tokio::spawn(async move {
        srv.start().await.expect("HTTP: admin api error");
        // kick off
        let client = reqwest::Client::builder()
            .add_root_certificate(cert)
            .build()
            .unwrap();
        let resp = client.get(url).send().await;
        assert!(resp.is_ok());
        let resp = resp.unwrap();
        assert!(resp.status().is_success());
        assert_eq!("/v1/health", resp.url().path());
    });
```

**Comment by sunisdown** [comment]
> Smells really good! Thank you man!

updated the test

**Comment by drmingdrmer** [comment]
> > Smells really good! Thank you man!
> 
> updated the test

History is erased by a force push...
Can not find out what has changed...

**Pull request merged**